openmc/model/surface_composite.py

```diff
@@ -0,0 +1,394 @@
+        Sides of the box
+    zmin, zmax : openmc.ZPlane
+        Sides of the box
+
+    """
+    _surface_names = ('xmin', 'xmax', 'ymin', 'ymax', 'zmin', 'zmax')
+
+    def __init__(self, v, a1, a2, a3, **kwargs):
+        vx, vy, vz = v
+        a1x, a1y, a1z = a1
+        a2x, a2y, a2z = a2
+        a3x, a3y, a3z = a3
+
+        # Only support boxes with axis-aligned vectors
+        if any(x != 0.0 for x in (a1y, a1z, a2x, a2z, a3x, a3y)):
+            raise NotImplementedError('Box macrobody with non-axis-aligned '
+                                      'vector not supported.')
+
+        # Determine each side of the box
+        if a1x > 0:
+            xmin, xmax = vx, vx + a1x
+        else:
+            xmin, xmax = vx + a1x, vx
+        if a2y > 0:
+            ymin, ymax = vy, vy + a2y
+        else:
+            ymin, ymax = vy + a2y, vy
+        if a3z > 0:
+            zmin, zmax = vz, vz + a3z
+        else:
+            zmin, zmax = vz + a3z, vz
+
+        # Create surfaces
+        self.xmin = openmc.XPlane(x0=xmin, **kwargs)
+        self.xmax = openmc.XPlane(x0=xmax, **kwargs)
+        self.ymin = openmc.YPlane(y0=ymin, **kwargs)
+        self.ymax = openmc.YPlane(y0=ymax, **kwargs)
+        self.zmin = openmc.ZPlane(z0=zmin, **kwargs)
+        self.zmax = openmc.ZPlane(z0=zmax, **kwargs)
+
+    def __neg__(self):
+        return (+self.xmin & -self.xmax &
+                +self.ymin & -self.ymax &
+                +self.zmin & -self.zmax)
+
+    def __pos__(self):
+        return (-self.xmin | +self.xmax |
+                -self.ymin | +self.ymax |
+                -self.zmin | +self.zmax)
+
+
+class XConeOneSided(CompositeSurface):
+    """One-sided cone parallel the x-axis
+
+    A one-sided cone is composed of a normal cone surface and an "ambiguity"
+    surface that eliminates the ambiguity as to which region of space is
+    included. This class acts as a proper surface, meaning that unary `+` and
+    `-` operators applied to it will produce a half-space. The negative side is
+    defined to be the region inside of the cone.
+
+    .. versionadded:: 0.12
+
+    Parameters
+    ----------
+    x0 : float, optional
+        x-coordinate of the apex. Defaults to 0.
+    y0 : float, optional
+        y-coordinate of the apex. Defaults to 0.
+    z0 : float, optional
+        z-coordinate of the apex. Defaults to 0.
+    r2 : float, optional
+        Parameter related to the aperature. Defaults to 1.
+    up : bool
+        Whether to select the side of the cone that extends to infinity in the
+        positive direction of the coordinate axis
+    **kwargs
+        Keyword arguments passed to underlying plane classes
+
+    Attributes
+    ----------
+    cone : openmc.XCone
+        Regular two-sided cone
+    plane : openmc.XPlane
+        Ambiguity surface
+    up : bool
+        Whether to select the side of the cone that extends to infinity in the
+        positive direction of the coordinate axis
```

Comment:
May want to mention that this corresponds to the positive half space of the plane so rotation doesn't confuse the user as to which way is "up".

Reply:
Good point, I'll add that to the description